### ISSUE TITLE
There should only be one internal triple name

### ISSUE THREAD

**Issue by glaubitz** [M680x0/M680x0-mono-repo#6]
The code currently allows both ```llvm::Triple::m680x0``` and  ```llvm::Triple::m68k``` as valid triple names.

However, as for any other architectures, there should be only one triple name and that should be the name of the backend, i.e. ```m680x0```.

We can still allow specifying the target on the command line using ```m68k```, similar to ```SystemZ``` which uses ```s390``` on the command line, but always ```SystemZ``` or ```systemz``` internally.

**Comment by glaubitz** [comment]
Suggested change:

```diff
diff --git a/clang/lib/Driver/ToolChains/Gnu.cpp b/clang/lib/Driver/ToolChains/Gnu.cpp
index 0394d99c6a7..50ceb0d64de 100644
--- a/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -2279,7 +2281,7 @@ void Generic_GCC::GCCInstallationDetector::AddDefaultGCCPrefixes(
       BiarchTripleAliases.append(begin(X86_64Triples), end(X86_64Triples));
     }
     break;
-  case llvm::Triple::m68k:
+  case llvm::Triple::m680x0:
     LibDirs.append(begin(M680x0LibDirs), end(M680x0LibDirs));
     TripleAliases.append(begin(M680x0Triples), end(M680x0Triples));
     break;
diff --git a/clang/lib/Sema/SemaDeclAttr.cpp b/clang/lib/Sema/SemaDeclAttr.cpp
index 00f7526c27e..1a670e04d19 100644
--- a/clang/lib/Sema/SemaDeclAttr.cpp
+++ b/clang/lib/Sema/SemaDeclAttr.cpp
@@ -6029,7 +6029,6 @@ static void handleInterruptAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
     handleMipsInterruptAttr(S, D, AL);
     break;
   case llvm::Triple::m680x0:
-  case llvm::Triple::m68k:
     handleM680x0InterruptAttr(S, D, AL);
     break;
   case llvm::Triple::x86:
diff --git a/llvm/include/llvm/ADT/Triple.h b/llvm/include/llvm/ADT/Triple.h
index ef8df9a0d44..ddb91b4f66a 100644
--- a/llvm/include/llvm/ADT/Triple.h
+++ b/llvm/include/llvm/ADT/Triple.h
@@ -75,7 +75,6 @@ public:
     tcele,          // TCE little endian (http://tce.cs.tut.fi/): tcele
     thumb,          // Thumb (little endian): thumb, thumbv.*
     thumbeb,        // Thumb (big endian): thumbeb
-    m68k,           // M680x0: Motorola 680x0 family
     m680x0,         // M680x0: Motorola 680x0 family
     x86,            // X86: i[3-9]86
     x86_64,         // X86-64: amd64, x86_64
diff --git a/llvm/lib/Support/Triple.cpp b/llvm/lib/Support/Triple.cpp
index c79a1fa0221..c8d73cf8c24 100644
--- a/llvm/lib/Support/Triple.cpp
+++ b/llvm/lib/Support/Triple.cpp
@@ -72,7 +72,6 @@ StringRef Triple::getArchTypeName(ArchType Kind) {
   case wasm32:         return "wasm32";
   case wasm64:         return "wasm64";
   case m680x0:         return "m680x0";
-  case m68k:           return "m68k";
   case x86:            return "i386";
   case x86_64:         return "x86_64";
   case xcore:          return "xcore";
@@ -152,8 +151,7 @@ StringRef Triple::getArchTypePrefix(ArchType Kind) {
   case riscv64:     return "riscv";
 
   case ve:          return "ve";
-  case m68k:
-  case m680x0:     return "m68k";
+  case m680x0:      return "m68k";
   }
 }
 
@@ -325,7 +323,6 @@ Triple::ArchType Triple::getArchTypeForLLVMName(StringRef Name) {
     .Case("renderscript64", renderscript64)
     .Case("ve", ve)
     .Case("m680x0", m680x0)
-    .Case("m68k", m68k)
     .Default(UnknownArch);
 }
 
@@ -425,8 +422,7 @@ static Triple::ArchType parseArch(StringRef ArchName) {
            "mips64r6", "mipsn32r6", Triple::mips64)
     .Cases("mips64el", "mipsn32el", "mipsisa64r6el", "mips64r6el",
            "mipsn32r6el", Triple::mips64el)
-    .Case("m680x0", Triple::m680x0)
-    .Case("m68k", Triple::m68k)
+    .Cases("m680x0", "m68k", Triple::m680x0)
     .Case("r600", Triple::r600)
     .Case("amdgcn", Triple::amdgcn)
     .Case("riscv32", Triple::riscv32)
@@ -672,8 +668,6 @@ static Triple::ObjectFormatType getDefaultFormat(const Triple &T) {
   case Triple::thumb:
   case Triple::x86:
   case Triple::x86_64:
-  case Triple::m680x0:
-  case Triple::m68k:
     if (T.isOSDarwin())
       return Triple::MachO;
     else if (T.isOSWindows())
@@ -696,6 +690,7 @@ static Triple::ObjectFormatType getDefaultFormat(const Triple &T) {
   case Triple::lanai:
   case Triple::le32:
   case Triple::le64:
+  case Triple::m680x0:
   case Triple::mips64:
   case Triple::mips64el:
   case Triple::mips:
@@ -1276,7 +1271,6 @@ static unsigned getArchPointerBitWidth(llvm::Triple::ArchType Arch) {
   case llvm::Triple::xcore:
   case llvm::Triple::wasm32:
   case llvm::Triple::m680x0:
-  case llvm::Triple::m68k:
     return 32;
 
   case llvm::Triple::aarch64:
@@ -1361,7 +1355,6 @@ Triple Triple::get32BitArchVariant() const {
   case Triple::x86:
   case Triple::xcore:
   case Triple::m680x0:
-  case Triple::m68k:
     // Already 32-bit.
     break;
 
@@ -1401,7 +1394,6 @@ Triple Triple::get64BitArchVariant() const {
   case Triple::tcele:
   case Triple::xcore:
   case Triple::m680x0:
-  case Triple::m68k:
     T.setArch(UnknownArch);
     break;
 
@@ -1518,7 +1510,6 @@ Triple Triple::getLittleEndianArchVariant() const {
   case Triple::sparcv9:
   case Triple::systemz:
   case Triple::m680x0:
-  case Triple::m68k:
 
   // ARM is intentionally unsupported here, changing the architecture would
   // drop any arch suffixes.
diff --git a/llvm/lib/Target/M680x0/TargetInfo/M680x0TargetInfo.cpp b/llvm/lib/Target/M680x0/TargetInfo/M680x0TargetInfo.cpp
index 89975a8a113..eef69789519 100644
--- a/llvm/lib/Target/M680x0/TargetInfo/M680x0TargetInfo.cpp
+++ b/llvm/lib/Target/M680x0/TargetInfo/M680x0TargetInfo.cpp
@@ -25,6 +25,4 @@ Target llvm::TheM680x0Target;
 extern "C" void LLVMInitializeM680x0TargetInfo() {
   RegisterTarget<Triple::m680x0, /*HasJIT=*/true> X(
       TheM680x0Target, "m680x0", "Motorola 68000 family", "M680x0");
-  RegisterTarget<Triple::m68k, /*HasJIT=*/true> K(
-      TheM680x0Target, "m68k", "Motorola 68000 family", "M680x0");
 }
```

**Comment by mshockwave** [comment]
Fixed by #7
